networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py

```diff
@@ -123,21 +126,26 @@ def get_vrf_context(self):
             return None
 
     def _get_vrf_context_gbp(self, context, router_id, port_db):
+        # TODO(tbachman): inefficient - should be fixed
         l3p_list = self.apic_driver.gbp_plugin.get_l3_policies(
-            context.elevated(), {'router': [router_id]})
-        kwargs = {'vrf_id': l3p_list[0]['id']}
-        details = self.apic_driver.get_vrf_details(context, **kwargs)
-        # The L3 out VLAN allocation uses UUIDs instead of names
-        details['vrf_name'] = details['l3_policy_id']
-        # get rid of VRF tenant -- not needed  for GBP
-        if details.get('vrf_tenant'):
-            details['vrf_tenant'] = None
-        return details
+            context.elevated(), filters={'routers': [router_id]})
+        for l3p in l3p_list:
+            if l3p.get('routers') and router_id in l3p['routers']:
+                break
+        if not l3p_list or not l3p:
+            return None
+        return {'vrf_id': l3p['id'],
+                'vrf_name': l3p['id'],
+                'vrf_tenant': None}
 
     def _get_vrf_context_neutron(self, context, router_id, port_db):
         router = self.l3_plugin.get_router(context, router_id)
         vrf_info = self.apic_driver.get_router_vrf_and_tenant(router)
         details = {}
+        if self.apic_driver.per_tenant_context:
+            details['vrf_id'] = router['tenant_id']
```

Comment:
don't think u need to do anything special here when per_tenant_context is true. The ML2 API get_router_vrf_and_tenant() should already take care of that for u. So u should be able to just use vrf_info['aci_name'] directly. Pls test though.